PS-9444 Release notes PS-5.7.44-52 (#426)
percona/psmysql-docs: docs/release-notes/5.7.44-52.md

Percona Server for MySQL 5.7.44-52 contains the following fixes.

## Bug fixes

### Security fixes

| Bug ID(s)                 | Description                                                                                                                                                   |
|---------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Bug #36492114             | Some combinations of optimizer hints did not function correctly.                                                                                             |
| Bug #36593265             | The AES_ENCRYPT() function did not always return a valid result.                                                                                             |
| Bug #36684463             | The internal function `my_convert_internal()` sometimes returned a pointer rather than the intended value.                                                   |

> | Bug #36593265             | Fixed an issue relating to FTS comparisons. While dumping tablespaces, `mysqldump` did not properly escape certain SQL statements in its output. Additionally, the dump now encloses the following identifiers within backticks: LOGFILE GROUP, TABLESPACE, and ENGINE. |

Reviewer: ```suggestion
| Bug #36210202             | Fixed an issue relating to FTS comparisons.
| Bug #36816986             | While dumping tablespaces, `mysqldump` did not properly escape certain SQL statements in its output. Additionally, the dump now encloses the following identifiers within backticks: LOGFILE GROUP, TABLESPACE, and ENGINE. |
```

# Percona Server for MySQL 5.7.44-52 (2024-12-05)

This release is part of {{post}}. Our [MySQL 5.7 Post-EOL Support] program delivers critical updates and stability for businesses that depend on this database version. Customers can access the full release through our private repository, receiving comprehensive binaries. Community members can [build the release] from publicly available source code, which we release at a later time.

We recommend that you upgrade to MySQL 8.0 or you can stay on 5.7; we'll support you either way.

> ## Release Information

Reviewer 2: ```suggestion
## Release information
```

Percona Server for MySQL 5.7.44-52 builds upon the functionality and bug fixes in MySQL 5.7.44 Community Edition. It adds enterprise-grade features and security enhancements developed by Percona.

Percona Server for MySQL 5.7.44-52 contains the following fixes.

## Bug fixes

### Security fixes

| Bug ID(s)                 | Details                                                                                                                                                   |
|---------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------|
| [Bug #36492114](https://github.com/mysql/mysql-server/commit/0f8002cf6ae)             | Some optimizer hint combinations did not function correctly.                                                                                             |
| [Bug #36593265](https://github.com/mysql/mysql-server/commit/c6997b13cfa)             | The AES_ENCRYPT() function returned incorrect results in some cases.                                                                                             |

> | [Bug #36684463](https://github.com/mysql/mysql-server/commit/de82d8ade0a)             | The internal function `my_convert_internal()` unexpectantly returned a pointer at times instead of the value.                                                   |

Reviewer: ```suggestion
| [Bug #36684463](https://github.com/mysql/mysql-server/commit/4ea181e95f0)             | The internal function `my_convert_internal()` unexpectantly returned a pointer at times instead of the value.                                                   |
```

|[Bug #36210202](https://github.com/mysql/mysql-server/commit/81a7f6fab54) | An issue related to FTS comparisons has been resolved.|

### Non-security fixes

| Bug Fix  | Details                                                                                          |
|---|----|
| [Bug #36559642](https://github.com/mysql/mysql-server/commit/5484c73368a)    | The server would unexpectedly exit when dropping an index and creating a full-text index in the same transaction. |
| [Bug #36600203](https://github.com/mysql/mysql-server/commit/de82d8ade0a)    | A memory leak in the MySQL client has been fixed.                                                             |
| [Bug #36871927](https://github.com/mysql/mysql-server/commit/2dd5e2ededb)    | The `DATABASE()` function now returns the correct output when used in `UNION` queries. The output from `DATABASE()` was truncated in earlier versions.   |

> ## Security enhancements

Reviewer: ```suggestion
## Third-party fixes[¶]
```

> This security update contains fixes and addresses the following vulnerabilities. To learn more about the vulnerabilities, see the following security advisories:

Reviewer: ```suggestion
The following Common Vulnerabilities and Exposures (CVE) records have been identified in third-party libraries:
```
